packages/sit-onyx/src/utils/forms.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * Extracts the first invalid validity type from the given HTML ValidityState.
+ * "valueMissing" is prioritized over other types to align with the default browser behavior.
+ */
+export const getFirstInvalidType = (validity: ValidityState) => {
+  // since the types are getters of the ValidityState we need to get the keys using "Object.getOwnPropertyDescriptors"
+  const availableValidityTypes = Object.entries(
+    Object.getOwnPropertyDescriptors(ValidityState.prototype),
+  )
+    .filter(([key, value]) => key !== "valid" && value.enumerable)
+    .map(([key]) => key) as Exclude<keyof ValidityState, "valid">[];
+
+  // sort valueMissing first to align with the default browser behavior
+  availableValidityTypes.sort((a, b) => {
+    if (a === "valueMissing") return -1;
+    if (b === "valueMissing") return 1;
+    return 0;
+  });
```

Comment:
technically this could simplify it

```suggestion
  // the error for an empty "required" value gets top priority and is returned first
  if (validity.valueMissing) return "valueMissing";

  // since the types are getters of the ValidityState we need to get the keys using "Object.getOwnPropertyDescriptors"
  const availableValidityTypes = Object.entries(
    Object.getOwnPropertyDescriptors(ValidityState.prototype),
  )
    .filter(([key, value]) => key !== "valid" && value.enumerable)
    .map(([key]) => key) as Exclude<keyof ValidityState, "valid">[];
```